samwise.js

```diff
@@ -1,19 +1,18 @@
 /**
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.